File: problems/0322.零钱兑换/0322-零钱兑换.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        int Max=amount+1;
        vector<int> dp(amount+1, Max);
        dp[0]=0;
        for(int i=1; i<(int)dp.size(); i++){
            for(int j=0; j<(int)coins.size(); j++){
                if(i>=coins[j]){
                    dp[i]=min(dp[i], dp[i-coins[j]]+1);
                }
            }
        }
        return dp[amount]>amount?-1:dp[amount];
    }
};

File: 0322.零钱兑换/0322-零钱兑换.js
/**
 * @param {number[]} coins
 * @param {number} amount
 * @return {number}
 */
var coinChange = function(coins, amount) {
    let dp=new Array(amount+1).fill(10010);
    
    dp[0]=0;
    for(let i=1; i<=amount; i++){
        for(let coin of coins){
            if(coin>i) continue;
            dp[i]=Math.min(dp[i-coin]+1, dp[i]);
        }
    }
    // console.log(dp);
    return dp[amount]===10010?-1:dp[amount];
};